tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -299,40 +303,56 @@ private bool UpdateRulesData()
 
         private void UpdateRuleStatus(IDictionary<string, bool> ruleStatus)
         {
-            if (ruleStatus != null && ruleStatus.Count > 0 && !_waf.ToggleRules(ruleStatus))
+            if (ruleStatus is { Count: > 0 } && !(_waf?.ToggleRules(ruleStatus) ?? false))
             {
                 Log.Debug($"_waf.ToggleRules returned false ({ruleStatus.Count} rule status entries)");
             }
         }
 
         private void UpdateStatus(bool fromRemoteConfig = false)
         {
-            lock (_settings)
+            if (_settings.Enabled)
```

Comment:
Is it safe to remove the lock here? 🤔 I wonder if `UpdateStatus` can be called concurrently (IIRC, it _can_)

Reply:
it _should_  be safe as RCM is not concurrent, so normally we can't have `UpdateStatus` called concurrently. What can be concurrent is creating Context(waf) per request and disposing it at the same time from an RCM event